internal/services/network/ip_group_resource.go

```diff
@@ -183,8 +183,25 @@ func resourceIpGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		}
 	}
 
-	d.Set("firewall_ids", getIds(resp.Firewalls))
-	d.Set("firewall_policy_ids", getIds(resp.FirewallPolicies))
+	var firewallIDs []string
+	for _, idStr := range getIds(resp.Firewalls) {
+		firewallID, err := azurefirewalls.ParseAzureFirewallIDInsensitively(idStr)
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall ID %q: %+v", idStr, err)
+		}
+		firewallIDs = append(firewallIDs, firewallID.ID())
+	}
+	d.Set("firewall_ids", firewallIDs)
+
+	var firewallPolicyIDs []string
+	for _, idStr := range getIds(resp.FirewallPolicies) {
+		policyID, err := firewallpolicies.ParseFirewallPolicyIDInsensitively(idStr)
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall Policy ID %q: %+v", idStr, err)
+		}
+		firewallPolicyIDs = append(firewallPolicyIDs, policyID.ID())
+	}
+	d.Set("firewall_policy_ids", firewallPolicyIDs)
```

Comment:
This issue still exists however - particularly in the Delete but also if the validation is wrong in the Create/Update/if refresh is disabled:

1. Update the `Create` function so that the `Parse` function raises an error when parsing fails
2. Update the `Update` function so that the `Parse` function raises an error when parsing fails
3. Update the `Delete` function so that the `Parse` function raises an error when parsing fails - but _also_ to load the values from the Azure API, since we shouldn't be using `d.Get` in the Delete function.

As such, can we update this?